node/src/components/consensus/highway_core/state.rs

```diff
@@ -566,8 +575,54 @@ impl<C: Context> State<C> {
                 return Err(VoteError::ValueAfterTerminalBlock);
             }
         }
-        // TODO: Validate against LNC.
-        Ok(())
+        match self.validate_lnc(wvote) {
+            Ok(_) => Ok(()),
+            Err(LncError::MissingEndorsement(vhash)) => {
+                panic!("Protocol state is missing endorsement for {:?}", &vhash)
```

Comment:
We shouldn't check for the endorsements in `validate_vote`; these should be resolved in `missing_dependency`.

`validate_vote` should just check whether the endorsement _claims_ that the message makes are enough to satisfy the LNC.

Reply:
This is more of a sanity check. We already download endorsements this vote announces in `missing_dependency`.

Reply:
I'd remove it; or panic.
I don't think we re-check here whether we have all cited _units_ either?

Reply:
I do panic here, I didn't want to panic in `validate_lnc` so that I can unit test it. I will remove it.

Reply:
Ah, I see. OK, I'm fine with either, I think.